packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -1693,3 +1697,379 @@ def test_no_tiering_occurs_when_custom_violating_query_selects_correct_columns_i
         # The ordering of DATA_NAME and COLL_NAME is flipped.
         columns = 'COLL_NAME, DATA_NAME, USER_NAME, USER_ZONE, DATA_REPL_NUM'
         self.do_incorrect_violating_query_test(columns)
+
+
+class test_executing_rules_as_rodsuser__issue_293(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.user1 = session.mkuser_and_return_session("rodsuser", "tolstoy", "tpass", lib.get_hostname())
+
+        self.tier0 = "ufs0"
+        self.tier1 = "ufs1"
+        self.tier0_time_in_seconds = 5
+
+        self.filename = "test_executing_rules_as_rodsuser__issue_293"
+        if not os.path.exists(self.filename):
+            lib.create_local_testfile(self.filename)
+
+        self.collection_path = self.user1.home_collection
+        self.object_path = "/".join([self.collection_path, self.filename])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand(['iqdel', '-a'])
+
+            lib.create_ufs_resource(admin_session, self.tier0, test.settings.HOSTNAME_2)
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier0, 'irods::storage_tiering::group', 'example_group', '0'])
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier0, 'irods::storage_tiering::time', str(self.tier0_time_in_seconds)])
+            admin_session.assert_icommand(
+                ['imeta', 'ls', '-R', self.tier0], 'STDOUT', [f'value: {self.tier0_time_in_seconds}', 'value: example_group'])
+
+            lib.create_ufs_resource(admin_session, self.tier1, test.settings.HOSTNAME_3)
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier1, 'irods::storage_tiering::group', 'example_group', '1'])
+            admin_session.assert_icommand(['imeta', 'ls', '-R', self.tier1], 'STDOUT', 'value: example_group')
+
+    @classmethod
+    def tearDownClass(self):
+        if os.path.exists(self.filename):
+            os.unlink(self.filename)
+
+        self.user1.__exit__()
+
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand(['iadmin', 'rmuser', self.user1.username])
+
+            admin_session.assert_icommand(['iqdel', '-a'])
+
+            admin_session.assert_icommand(['iadmin', 'rmresc', self.tier0])
+            admin_session.assert_icommand(['iadmin', 'rmresc', self.tier1])
+            admin_session.assert_icommand(['iadmin', 'rum'])
+
+    def tearDown(self):
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand(["ichmod", "-M", "own", admin_session.username, self.object_path])
+            admin_session.assert_icommand(["irm", "-f", self.object_path])
```

Comment:
teardowns usually... just run_ rather than assert_ ?

Reply:
Changed all `tearDown` and `tearDownClass` uses of `assert_icommand` to `run_icommand` for the new tests.